lib/archethic/mining/distributed_workflow.ex

```diff
@@ -939,6 +978,86 @@ defmodule Archethic.Mining.DistributedWorkflow do
     })
   end
 
+  defp request_replication_validation(
+         context = %ValidationContext{transaction: tx, coordinator_node: coordinator_node},
+         node_public_key
+       ) do
+    storage_nodes = ValidationContext.get_chain_replication_nodes(context)
+
+    Logger.info(
+      "Send validated transaction to #{Enum.map_join(storage_nodes, ",", &Node.endpoint/1)} for replication's validation",
+      transaction_address: Base.encode16(tx.address),
+      transaction_type: tx.type
+    )
+
+    validated_tx = ValidationContext.get_validated_transaction(context)
+
+    message = %ValidateTransaction{
+      transaction: validated_tx
+    }
+
+    results =
+      Task.Supervisor.async_stream_nolink(
+        Archethic.TaskSupervisor,
+        storage_nodes,
+        fn node ->
+          {P2P.send_message(node, message), node}
+        end,
+        ordered: false,
+        on_timeout: :kill_task,
+        timeout: Message.get_timeout(message) + 2000
+      )
+      |> Stream.filter(&match?({:ok, _}, &1))
```

Comment:
Here this filter may cause problem.
If a node do not respond to the message, this filter removes it from the list. Then we add the remaining nodes public key in the validation context. But as there is one node never added to the validation context, the function `ValidationContext.enough_replication_validations?` will never return true as a node is missing.
So if a node do not respond to the `ValidateTransaction` message, we already now that the transaction replication will never happen.
There is 2 way to think with this problem,
- We want all storage node to respond this message, so if one do not respond we can directly enter in replication_error.
- We want to continue even if a storage node do not respond this message, so instead of sending the public of the responding storage node, the validation node can inform the other validation nodes that it's sub shard is ok for replication. And then we start the replication if all the validation nodes told their sub shard is ok.

Reply:
Nicely seen..
Indeed this is one pitfall of this strict approach. 
Indeed in the current implementation the shard would be invalidated if one replication node which was supposed to be available does not respond. Causing the other nodes to await the timeout. While this seems acceptable in term of transaction validation point of view, we might to inform the user about the transaction failure earlier. 

We have to support fault tolerance in all cases. The shard size can be viewed as an overbooking against the locally available nodes. 
So the 2nd approach might be more suited for the case, as the unavailability of the node is an hazard, considered as benignly fault and not intentional/malicious behavior.